axelor-sale/src/main/resources/views/AppSale.xml

```diff
@@ -25,5 +26,9 @@
 		<view type="grid" name="sale-config-grid" />
 		<view type="form" name="sale-config-form"/>
 	</action-view>
-	
+
+  <action-attrs name="action-app-sale-attrs-manage-sale-order-version-onchange">
+    <attribute name="value" for="generatePdfDuringSaleOrderFinalization" expr="true"/>
```

Comment:
better to define an expression using manageSaleORderVersion boolean, instead of "true". Else, the boolean generatePdfDuringSaleOrderFinalization will never be reseted. --> printingOnSOFinalization ?

Also, use a shorter word for this new boolean to avoid database index issue or URL issue due to too long term.

Reply:
done